src/dmd/denum.d

```diff
@@ -415,6 +415,11 @@ extern (C++) final class EnumMember : VarDeclaration
         if (errors)
             return new ErrorExp();
         checkDisabled(loc, sc);
+
+        if (depdecl && !depdecl._scope)
+            depdecl._scope = sc;
```

Comment:
Took me a while to see that `DeprecatedDeclaration._scope` is used here:
https://github.com/dlang/dmd/blob/58fde675fb69ea57da8902ddbabf87cf8b74e1f9/src/dmd/dsymbolsem.d#L410-L428
which is called from here:
https://github.com/dlang/dmd/blob/ce5e17727e43d60991091fc96fbe5fc0d7435334/src/dmd/dsymbol.d#L300-L324